packages/core/src/util.ts

```diff
@@ -111,7 +111,7 @@ export function create_chunk_key_encoder({
 }: ArrayMetadata["chunk_key_encoding"]): (chunk_coords: number[]) => string {
 	if (name === "default") {
 		return (chunk_coords) =>
-			["c", ...chunk_coords].join(configuration.separator);
+			["c", ...chunk_coords].join(configuration?.separator || "/");
 	}
 	if (name === "v2") {
```

Comment:
Wait, but when the field is `v2` the `configuration` is required?

Reply:
This is `"name": "v2"` within v3 spec (confused me for a sec)! At https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html#chunk-encoding my understanding is that the `configuration` is optional regardless of the name. If the name is `v2` then the default separator is `.`